webhooks-extension/docs/GettingStarted.md

```diff
@@ -0,0 +1,161 @@
+> Webhooks allow external services to be notified when certain events happen. When the specified events happen, we’ll send a POST request to each of the URLs you provide. 
+
+So a 'webhook' is an outbound HTTP POST from GitHub to a given URL. It contains a rich payload in JSON format describing what just happened in Git. This might be, 'A Pull Request was opened', 'A commit was merged to a branch', or many other events. 
+
+It's important to understand that webhooks are outbound HTTP requests from GitHub to a given URL. This URL must be DNS-resolvable and IP-reachable from GitHub. If you are using `github.com` then the target URL must be on the external Internet. It is not generally possible to send a webhook from public `github.com` to a target (normally an Ingress endpoint on a Kubernetes cluster) behind a firewall. You should be using GitHub Enterprise today (or your own GitLab install in the future) if you wish to use webhooks to trigger Tekton Pipelines in a Kubernetes cluster situated behind a firewall. 
+
+## What does the Tekton Dashboard's Webhooks Extension do?
+
+The webhooks extension helps you to create a relationship between a git repository and a Tekton pipeline. This pipeline will be triggered when certain changes happen in the associated repository. Currently these changes are, 
+- A commit is merged into a branch, and
+- A Pull Request is created or modified
+
+See https://github.com/tektoncd/experimental/tree/master/webhooks-extension#notes-on-using-the-webhooks-extension for more details on how a webhook payload is turned into a `PipelineRun` and the necessary set of accompanying `PipelineResources`. 
+
+## Setting up a webhook from scratch
+
+Several steps must be completed in order to set up a working webhook. Pipeline definitions must be imported, and credentials set up to permit those pipelines to function. Credentials are typically required for GitHub, to create webhooks and clone repositories, and for a docker image registry to push images to. 
+
+Tekton PipelineRuns gain access to git and docker registry credentials via their associated service account. 
+- git credentials must be stored in a secret with an annotation, `tekton.dev/git-0: [url to git repository]`
+- docker registry credentials must be stored in a secret with an annotation, `tekton.dev/docker-0: [url to docker registry]`
+- credentials are located via the service account used by a given Tekton TaskRun or PipelineRun
+See the [Tekton documentation](https://github.com/tektoncd/pipeline/blob/master/docs/auth.md#basic-authentication-git) for more information. 
+
+Before you start creating Tekton resources, decide which Kubernetes namespace(s) you wish to execute pipelines in, and which service accounts will be used to execute them. Ensure that each service account has sufficient permissions under Kubernetes [Role-Based Access Control (RBAC)](https://kubernetes.io/docs/reference/access-authn-authz/rbac/) to execute your intended pipelines.
+
```

Comment:
We could mention using the `tekton-webhooks-extension` SA, that's powerful and if users wish to use their own, they can indeed go about making their own in this namespace to use (let's be honest, most will just use this as it's provided at install time, for convenience)

Reply:
> If you're just testing, the `tekton-webhooks-extension` service account in the `tekton-pipelines` namespace can be used to run a wide range of pipelines by virtue of the `tekton-webhooks-extension-minimal` role and binding.